File: duelink/arduino/DUELink/Digital.cpp
#ifdef ARDUINO
#include "Arduino.h"
#endif

#include "digital.h"

  bool DigitalController::Read(int pin, int pull) {
    char cmd[32];
    sprintf(cmd, "dread(%d,%d)", pin, pull);
    DUELinkTransport::Response result = m_pTransport->execute(cmd);
    if (result.success) return atoi(result.result.c_str());
    return false;
  }

  void DigitalController::Write(int pin, int state){
    char cmd[32];
    sprintf(cmd, "dwrite(%d,%d)", pin, state);
    m_pTransport->execute(cmd);
  }
  
  
  
  

File: duelink/micropython/duelink/digital.py

class DigitalController:    

    def __init__(self, transport):
        self.transport = transport

    def Read(self, pin, pull):
        r, s = self.execute(f"dread({pin},{pull})")
        if s:
            return int(r, 10)        

    def Write(self, pin, value):
        self.transport.execute(f"dwrite({pin},{value})")
